sound/soc/sof/ipc4-topology.h

```diff
@@ -82,6 +84,26 @@ struct sof_ipc4_pipeline {
 	struct sof_ipc4_msg msg;
 };
 
+/**
+ * struct sof_ipc4_multi_pipeline_info - multi pipeline trigger IPC info
+ * @pipelines: Array of pipeline widgets to be triggered
+ * @count: number of pipelines to be triggered
+ */
+struct sof_ipc4_multi_pipeline_info {
+	struct snd_sof_widget *pipelines[SOF_IPC4_TRIGGER_MAX_PIPELINE_COUNT];
+	uint32_t count;
+};
+
+/**
+ * struct sof_ipc4_multi_pipeline_data - multi pipeline trigger IPC data
+ * @count: Number of pipelines to be triggered
+ * @pipeline_ids: Arrays of IDs of the pipelines to be triggered
+ */
+struct ipc4_pipeline_set_state_data {
```

Comment:
these are sent to the firmware, so need to be packed?

Reply:
yes, thanks for catching that

Reply:
How can a compiler pack a flexible struct?